src/pass/storage_rewrite.cc

```diff
@@ -777,6 +780,23 @@ class StoragePlanRewriter : public IRMutator {
     return e;
   }
 
+  // check if op can reuse e's buffer
+  bool CanReuse(const StorageEntry *e,
+                const Allocate* op,
```

Comment:
The reuse rule of constant buffer and non-constant one may differ for dynamic and static known size case and that is why we didn't isolate this. Consider move part of logic back to loops and add detailed comment